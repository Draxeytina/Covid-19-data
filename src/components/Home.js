import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getData } from '../redux/data';
import Data from './Data';
import Filter from './Filter';
import './home.scss';

export default function Home() {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.data);
  const [filtered, setFiltered] = useState([]);
  const [activeCountry, setActiveCountry] = useState('All');

  useEffect(() => {
    if (!data.length) { dispatch(getData()); }
  }, []);

  return (
    <>
      <div className="home">
        Welcome
      </div>

      <Filter
        data={data}
        setFiltered={setFiltered}
        activeCountry={activeCountry}
        setActiveCountry={setActiveCountry}
      />
      <div className="list-container-oval">
        {
          (activeCountry === 'Top World Cases') ? (
            <div>
              <iframe
                title="World Top Data"
                src="https://covid-19.dataflowkit.com/assets/widget/c19-top-widget.html"
                width="90%"
                height="300px"
                frameBorder="0"
              />
            </div>
          ) : (
            <div className="regions-container">
              <ul>
                {filtered.length === 0 && (

                  data.map((data) => (
                    <li key={data.id}>
                      <Data
                        countryName={data.Country_text}
                        totalCases={data['Total Cases_text']}
                        totalDeaths={data['Total Deaths_text']}
                        totalRecoveries={data['Total Recovered_text']}
                      />
                      <Link
                        to={`/covidData/${data.id}`}
                      >
                        <button type="button">View Statistics</button>
                      </Link>
                    </li>
                  ))
                )}
                {filtered.length !== 0 && (

                  filtered.map((data) => (
                    <li key={data.id}>
                      <Data
                        countryName={data.Country_text}
                        totalCases={data['Total Cases_text']}
                        totalDeaths={data['Total Deaths_text']}
                        totalRecoveries={data['Total Recovered_text']}
                      />
                      <Link
                        to={`/covidData/${data.id}`}
                      >
                        <button type="button">View Statistics</button>
                      </Link>
                    </li>
                  ))
                )}
              </ul>
            </div>
          )
        }
      </div>
    </>
  );
}
