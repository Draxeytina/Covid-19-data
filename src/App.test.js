import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from './redux/configureStore';
import Home from './components/Home';

const renderMissions = (
  <Provider store={store}>
    <Home />
  </Provider>
);

describe('Home page', () => {
  test('renders home page', () => {
    render(renderMissions);
    const header = screen.getByText('Welcome');
    expect(header).toBeInTheDocument();
  });
});

describe('Home page snapshot', () => {
  it('renders the home page and makes a snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
