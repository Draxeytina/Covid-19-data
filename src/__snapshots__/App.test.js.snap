// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page snapshot renders the home page and makes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h1
      class="welcome-text"
    >
      Welcome
    </h1>
  </div>
  <div
    class="filter"
  >
    Select Region : 
    <select
      class="categoryInput"
      name="country"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Southern African Countries"
      >
        Southern African Countries
      </option>
      <option
        value="Northen African Countries"
      >
        Northen African Countries
      </option>
      <option
        value="Western African Countries"
      >
        Western African Countries
      </option>
      <option
        value="Eastern African Countries"
      >
        Eastern African Countries
      </option>
      <option
        value="Middle African Countries"
      >
        Middle African Countries
      </option>
      <option
        value="Top World Cases"
      >
        Top World Cases
      </option>
    </select>
  </div>
  <div
    class="list-container-oval"
  >
    <div>
      <ul
        class="regions-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
